### PR TITLE
Update rack-test dependency constraint

### PR DESCRIPTION
**Pull request by dwightwatson**
### Summary

This updates the dependency constraint for `rack-test` in `actionpack`. `rack-test` is the only gem that appears as outdated when running `bundle outdated` on a fresh installation of Rails.

I've checked the changelog for `rack-test` and it doesn't appear to have breaking changes, and the `actionpack` tests passed locally for me after the change.

### Other Information

This is my first Rails PR so I hope I'm doing it right - wasn't sure whether I should commit the updated `Gemfile.lock` as well or if I needed to update the changelog. Happy to be pointed in the right direction for any changes.

### CODE REVIEW

File: actionpack/actionpack.gemspec [hunk 1]
```diff
@@ -27,7 +27,7 @@ Gem::Specification.new do |s|
   s.add_dependency "activesupport", version
 
   s.add_dependency "rack",      "~> 2.0"
-  s.add_dependency "rack-test", "~> 0.6.3"
+  s.add_dependency "rack-test", "~> 0.7.0"
```
Comment (rafaelfranca) [inline actionpack/actionpack.gemspec]: Is there any reason to disallow 0.6.3? If not this should be `s.add_dependency "rack-test", ">= 0.6.3"`

**Review by rafaelfranca — CHANGES_REQUESTED** [review]

File: actionpack/actionpack.gemspec (hunk 1 above)
Comment (dwightwatson) [inline actionpack/actionpack.gemspec]: Nope, that's fair - updated. Thank you.

**Comment by rafaelfranca** [comment]
Cloud you squash your commits?

**Comment by dwightwatson** [comment]
Of course, done.

**Comment by mvz** [comment]
This is not part of 5.1.3 😞

**Comment by rafaelfranca** [comment]
@mvz why do you need it in 5.1? Is there anything in rack-test that require us to backport?

**Comment by mvz** [comment]
@rafaelfranca only for the reason given for this PR:

>  rack-test is the only gem that appears as outdated when running bundle outdated on a fresh installation of Rails.

**Comment by rafaelfranca** [comment]
Done

**Comment by mvz** [comment]
Awesome, thanks!

**Comment by morgoth** [comment]
This caused regression in new release https://github.com/rails/rails/issues/30570 because of https://github.com/rack-test/rack-test/issues/200

**Pull request merged**